### PR TITLE
Block Hooks API: Update doc block for block_core_navigation_insert_hooked_blocks

### PR DESCRIPTION
**Pull request by tjcafferkey**
## What?

Updating doc block for `block_core_navigation_insert_hooked_blocks` used in the core Navigation block

## Why?

It was caught and highlighted during the package updates for 6.5 here and requires a PR in GB to fix https://github.com/WordPress/wordpress-develop/pull/5922#discussion_r1469460206

## How?

It's implementing suggested changes which are directly copied from Gutenberg.

### CODE REVIEW

File: packages/block-library/src/navigation/index.php [hunk 1]
```diff
@@ -1354,8 +1354,8 @@ function block_core_navigation_get_most_recently_published_navigation() {
  * children, the `wp_navigation` post's `_wp_ignored_hooked_blocks` meta is checked to see if any
  * of those hooked blocks should be exempted from insertion.
  *
- * @param array   $inner_blocks Parsed inner blocks of a Navigation block.
- * @param WP_Post $post         `wp_navigation` post object corresponding to the block.
+ * @param array        $inner_blocks Parsed inner blocks of a Navigation block.
+ * @param WP_Post|null $post         Optional. `wp_navigation` post object corresponding to the block. Default null.
```
Comment (ockham) [inline packages/block-library/src/navigation/index.php]: I briefly tried to glance over the callsites for this function. Is it possible that we eliminated all cases where `$post` could've been omitted? (If yes, we might want to make the argument mandatory.)

Comment (tjcafferkey) [reply packages/block-library/src/navigation/index.php]: Good spot, I've double checked this and the filter usage its attached to and you're right. I've updated here https://github.com/WordPress/gutenberg/pull/58378/commits/ecd6c3f73563402694d1ba50c7bb3c2ab1d96864

File: packages/block-library/src/navigation/index.php [hunk 2]
```diff
@@ -1355,10 +1355,10 @@ function block_core_navigation_get_most_recently_published_navigation() {
  * of those hooked blocks should be exempted from insertion.
  *
  * @param array   $inner_blocks Parsed inner blocks of a Navigation block.
- * @param WP_Post $post         `wp_navigation` post object corresponding to the block.
+ * @param WP_Post $post        `wp_navigation` post object corresponding to the block. Default null.
```
Comment (ockham) [inline packages/block-library/src/navigation/index.php]:
```suggestion
 * @param WP_Post $post        `wp_navigation` post object corresponding to the block.
```

😅

**Review by ockham — APPROVED** [review]
Thank you!

File: packages/block-library/src/navigation/index.php [hunk 3]
```diff
@@ -1355,10 +1355,10 @@ function block_core_navigation_get_most_recently_published_navigation() {
  * of those hooked blocks should be exempted from insertion.
  *
  * @param array   $inner_blocks Parsed inner blocks of a Navigation block.
- * @param WP_Post $post         `wp_navigation` post object corresponding to the block.
+ * @param WP_Post $post        `wp_navigation` post object corresponding to the block.
```
Comment (youknowriad) [inline packages/block-library/src/navigation/index.php]: The arguments are not aligned now no?

Comment (tjcafferkey) [reply packages/block-library/src/navigation/index.php]: Fixed in https://github.com/WordPress/gutenberg/pull/58378/commits/14e856c621cf57df45b46b6b15e842daf1479d94

Comment (tjcafferkey) [reply packages/block-library/src/navigation/index.php]: Fixed in https://github.com/WordPress/gutenberg/pull/58378/commits/14e856c621cf57df45b46b6b15e842daf1479d94

**Comment by cbravobernal** [comment]
I just cherry-picked this PR to the release/17.6 branch to get it included in the next release: ba74bfcb46

**Pull request merged**